
# Project Planning Stage(Individual)


In [1]:
library(tidyverse)
library(lubridate)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [2]:
url<- "https://drive.google.com/uc?export=download&id=1Mw9vW0hjTJwRWx0bDXiSpYsO3gKogaPz"
players_data <- read_csv(url)

head(players_data)

Rows: 196 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, age
lgl (3): subscribe, individualId, organizationName

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender age individualId organizationName
1 Morgan    Male    9  NA           NA              
2 Christian Male   17  NA           NA              
3 Blake     Male   17  NA           NA              
4 Flora     Female 21  NA           NA              
5 Kylie     Male   21  NA           NA              
6 Adrian    Female 17  NA           NA

## 1. Data Description

This dataset contains information about a Minecraft research server, and the data is collected by analyzing player gameplay, speech, and key presses. The dataset has 196 rows and 9 columns. This means there are 196 observations and 9 variables in the dataset. Each row represents a single player. Out of the 9 variables, 4 are characters, 2 are numerical, and 3 are logical.

#### Summary Statistics

In [20]:
players_data|>
summarize(across(
         c(played_hours, age),
    list(
        mean= ~ round(mean(.x),2),
        max = ~ round(max(.x),2),
        min= ~ round(min(.x),2),
        sd= ~ round(sd(.x),2))))
summarize(players_data,
          sum_played_hours= sum(played_hours))

subscribe_count <- players_data|>
count(subscribe)
experience_count <- players_data|>
  count(experience)
gender_count <- players_data |>
  count(gender)
age_count <- players_data |>
  count(age)
played_hours_count <- players_data |>
  count(played_hours)

subscribe_count
experience_count
gender_count
age_count
played_hours_count

played_hours_mean played_hours_max played_hours_min played_hours_sd age_mean
1 5.85              223.1            0                28.36           21.28   
  age_max age_min age_sd
1 99      8       9.71

sum_played_hours
1 1145.8

subscribe n  
1 FALSE      52
2  TRUE     144

experience n 
1 Amateur    63
2 Beginner   35
3 Pro        14
4 Regular    36
5 Veteran    48

gender            n  
1 Agender             2
2 Female             37
3 Male              124
4 Non-binary         15
5 Other               1
6 Prefer not to say  11
7 Two-Spirited        6

age n 
1   8   1
2   9   1
3  10   1
4  11   1
5  12   1
6  14   2
7  15   2
8  16   3
9  17  75
10 18   7
11 19   7
12 20  14
13 21  18
14 22  15
15 23  13
16 24  10
17 25   3
18 26   4
19 27   3
20 28   2
21 32   2
22 33   1
23 37   1
24 38   1
25 42   1
26 44   1
27 45   1
28 46   1
29 49   1
30 50   1
31 91   1
32 99   1

played_hours n 
1    0.0        85
2    0.1        34
3    0.2        10
4    0.3         5
5    0.4         5
6    0.5         4
7    0.6         5
8    0.7         3
9    0.8         2
10   0.9         1
11   1.0         6
12   1.1         1
13   1.2         2
14   1.4         1
15   1.5         2
16   1.6         1
17   1.7         1
18   1.8         2
19   2.0         1
20   2.1         1
21   2.2         1
22   2.3         1
23   2.7         2
24   2.9         1
25   3.5         1
26   3.6         1
27   3.8         1
28   4.0         1
29   5.6         1
30   7.1         1
31  12.5         1
32  17.2         1
33  18.5         1
34  23.7         1
35  30.3         1
36  32.0         1
37  48.4         1
38  53.9         1
39  56.1         1
40 150.0         1
41 178.2         1
42 218.1         1
43 223.1         1

#### Summary

| Variable| Type| Description| Potential Errors| Observations|
|---------|------|-----------|-----------------|--------------|
|experience|chr|Players Gaming Experience Level|none|5 categories- Amateur = most common|
|hashedEmail|chr|Players Emails|Possible duplicates- hard to tell at first glance|Anonymous|
|name|chr|Players Name|none|Likely not usefulA
|gender|chr|Players Gender|none|Most common - Men|
|played_hours|dbl|Amount of Hours played|none|Highly Fluctuates|
|age|dbl|Players Age|none|Range 8-99|
|subscribe|lgl|If Players are subscribed|none|144 are subscribed|
|individualID|lgl|invalid|All values are N/A|Cannot be used|
|organizationName|lgl|invalid|All values are N/A|Cannot be used|

Initial Observations 
- played_hours varies a lot between players
- Age also has a wide range
- Two columns are completing missing values
- hasedEmail and name are identifiers are will not be used in the predictions